wasm-thumbnail/src/lib.rs

```diff
@@ -32,11 +32,21 @@ pub extern "C" fn resize_and_pad(
     let slice: &[u8] = unsafe { std::slice::from_raw_parts(pointer, length) };
 
     let mut out: Vec<u8> = Vec::with_capacity(nsize);
-    // Reserve space at the start for length header
-    out.extend_from_slice(&[0, 0, 0, 0]);
 
-    if let Ok(thumbnail_len) = _resize_and_pad(slice, &mut out, nwidth, nheight, nsize, nquality) {
-        out.splice(..4, thumbnail_len.to_be_bytes().iter().cloned());
+
+
+    let mut mquality = nquality;
+
+    while mquality >= 15 {
+        // Reserve space at the start for length header
+        out.extend_from_slice(&[0, 0, 0, 0]);
```

Comment:
Doesn't repeating this line mean we'll be adding these 4 bytes each time round the loop?  So if we had 3 attempts to get the size down we'd have 12 bytes added?

Or maybe not - I'm not 100% sure what `extend_from_slice` actually does.

It'd be good to have evidence that this generates the correct output for an image that was failing previously - ideally in an automated test but if the setup of this project makes that too difficult then showing manual execution would be an ok fallback.

Reply:
Correct, will fix

Reply:
In practice, unless I reallocated the (now Cursor), the image would corrupt and the wasm would crash. In my tests, reallocating on failure of resize resulted in the image being properly resized. I attempted to just reset the position to the beginning, but no dice. I could try a zeroing out of the underlying vector, but that introduces a different tradeoff.